### ISSUE TITLE
gopls: automated issue report (initialization)

### ISSUE THREAD

**Issue by gobins** [golang/vscode-go#318]
Couldn't start client gopls

```
The language server is not able to serve any features. Initialization failed: Error: json: cannot unmarshal object into Go struct field .capabilities.textDocument.publishDiagnostics.tagSupport of type bool.
```

**Comment by hyangah** [comment]
@gobins Thanks for the info with the error detail. 

Seems like VS Code sent initialization message in the format `gopls` doesn't understand. Can you please capture the gopls message log so we can see how the message looks like? Add the following in the settings.json, 

```
"go.languageServerFlags": [
  "-rpc.trace"
]
```

and the trace will appear in the output panel (Command Palette -> View: Toggle Output -> select `gopls (server)` from the dropdown). 

@stamblerre @pjweinbgo `tagSupport` should be an object type, but seems like we got a boolean. We will see more details once we get the trace, but is it expected to fail if unrecognized field is detected?

**Comment by pjweinbgo** [comment]
You mean that vscode is sending a boolean? And then gopls is failing to
unmarshal?

**Comment by stamblerre** [comment]
@gobins: What versions of VS Code and VS Code Go are you using? My guess is that the issue is an old language client. You can find out the versions by running `code --version` and `code --list-extensions --show-versions`.

**Comment by gobins** [comment]
Thanks @stamblerre , the issue was fixed after I upgraded it to v0.4.3.
